simconvoi.cc

```diff
@@ -5382,8 +5395,14 @@ bool convoi_t::is_waiting_for_coupling() const {
 }
 
 bool convoi_t::check_electrification() {
-	is_electric = true;
+	is_electric = false;
 	convoihandle_t c = find_most_parent_convoi();
+	while(  c.is_bound()  &&  !is_electric  ) {
```

Comment:
`find_most_parent_convoi` は計算コストが高いので、一度だけ実行するようにしましょう。
```suggestion
	is_electric = false;
	const convoihandle_t most_parent_convoi = find_most_parent_convoi();
	convoihandle_t c = most_parent_convoi;
	while(  c.is_bound()  &&  !is_electric  ) {
```